### PR TITLE
[SCFToCalyx] Build control for nested if operation

### PR DESCRIPTION
**Pull request by jiahanxie353**
This patch supports `BuildControl` for nested `scf::If` operations

### CODE REVIEW

**Comment by rachitnigam** [comment]
PR needs description. Please add one and then tag reviewers!

File: lib/Conversion/SCFToCalyx/SCFToCalyx.cpp [hunk 1]
```diff
@@ -1551,8 +1551,13 @@ class BuildControl : public calyx::FuncOpPartialLoweringPattern {
             rewriter.create<calyx::SeqOp>(ifOp.getThenRegion().getLoc());
         auto *thenSeqOpBlock = thenSeqOp.getBodyBlock();
 
-        rewriter.setInsertionPointToEnd(thenSeqOpBlock);
+        auto *thenBlock = &ifOp.getThenRegion().front();
+        LogicalResult res = buildCFGControl(path, rewriter, thenSeqOpBlock,
```
Comment (cgyurgyik) [inline lib/Conversion/SCFToCalyx/SCFToCalyx.cpp:1555]: Just to make sure I'm understanding correctly: your code is only supporting one level of nesting of `if` statements?

Comment (jiahanxie353) [reply lib/Conversion/SCFToCalyx/SCFToCalyx.cpp:1555]: Arbitrary level because `buildCFGControl` is a recursive function

Comment (cgyurgyik) [reply lib/Conversion/SCFToCalyx/SCFToCalyx.cpp:1555]: Ah got it, so initially it was just ignoring the `then` and `else` blocks of the `if` control.

**Review by cgyurgyik — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**